tests/sentry/workflow_engine/migration_helpers/test_migrate_rule.py

```diff
@@ -406,3 +295,294 @@ def test_delete_issue_alert(self):
+            filter_match="any",
+            action_data=self.action_data,
+        )
+
+        IssueAlertMigrator(issue_alert, self.user.id).run()
+        assert DataCondition.objects.all().count() == 1
+        dc = DataCondition.objects.get(type=Condition.REGRESSION_EVENT)
+        assert dc.condition_group.logic_type == DataConditionGroup.Type.ANY_SHORT_CIRCUIT
+
+    def test_run__every_event_condition__all(self):
+        conditions = [
+            {"id": EveryEventCondition.id},
+            {"id": RegressionEventCondition.id},
+        ]
+        issue_alert = self.create_project_rule(
+            condition_data=conditions,
+            action_match="all",
+            filter_match="any",
+            action_data=self.action_data,
+        )
+
+        IssueAlertMigrator(issue_alert, self.user.id).run()
+        assert DataCondition.objects.all().count() == 1
+        dc = DataCondition.objects.get(type=Condition.REGRESSION_EVENT)
+        assert dc.condition_group.logic_type == DataConditionGroup.Type.ALL
+
+    def test_dry_run(self):
+        IssueAlertMigrator(self.issue_alert, self.user.id, is_dry_run=True).run()
+
+        assert not AlertRuleWorkflow.objects.filter(rule=self.issue_alert).exists()
+        assert not AlertRuleDetector.objects.filter(rule=self.issue_alert).exists()
+
+        assert Workflow.objects.all().count() == 0
+        assert Detector.objects.all().count() == 0
+        assert DataConditionGroup.objects.all().count() == 0
+        assert DataCondition.objects.all().count() == 0
+        assert Action.objects.all().count() == 0
+
+    def test_dry_run__already_exists(self):
+        IssueAlertMigrator(self.issue_alert, self.user.id).run()
+
+        with pytest.raises(Exception):
+            IssueAlertMigrator(self.issue_alert, self.user.id, is_dry_run=True).run()
+
+        issue_alert_workflow = AlertRuleWorkflow.objects.get(rule=self.issue_alert)
+        issue_alert_detector = AlertRuleDetector.objects.get(rule=self.issue_alert)
+        Workflow.objects.get(id=issue_alert_workflow.workflow.id)
+        Detector.objects.get(id=issue_alert_detector.detector.id)
+
+    @patch(
+        "sentry.workflow_engine.migration_helpers.issue_alert_migration.enforce_data_condition_json_schema"
+    )
+    def test_dry_run__data_condition_validation_fails(self, mock_enforce):
+        mock_enforce.side_effect = ValidationError("oopsie")
+
+        with pytest.raises(ValidationError):
+            IssueAlertMigrator(self.issue_alert, self.user.id, is_dry_run=True).run()
+
+        assert not AlertRuleWorkflow.objects.filter(rule=self.issue_alert).exists()
+        assert not AlertRuleDetector.objects.filter(rule=self.issue_alert).exists()
+
+        assert Workflow.objects.all().count() == 0
+        assert Detector.objects.all().count() == 0
+        assert DataConditionGroup.objects.all().count() == 0
+        assert DataCondition.objects.all().count() == 0
+        assert Action.objects.all().count() == 0
```

Comment:
nit: could be nice to put this repeated stuff into some function

Reply:
done ty!